deployability/modules/testing/tests/helpers/agent.py

```diff
@@ -607,3 +619,121 @@ def agent_status_report(wazuh_api: WazuhAPI) -> dict:
+
+    @staticmethod
+    def __parse_amazon(cmd_output: str) -> str:
+        pattern = r"Amazon Linux release (\d+)\.\d+\.\d+"
+        match = re.search(pattern, cmd_output)
+        return  match.group(1) if match else ""
+
+    @staticmethod
+    def __parse_rhel(cmd_output: str) -> str:
+        pattern = r".* ([0-9]{1,2})\.*[0-9]{0,2}.*"
+        match = re.search(pattern, cmd_output)
+        return  match.group(1) if match else ""
+
+    @staticmethod
+    def __parse_ubuntu(cmd_output: str) -> str:
+        pattern = r'VERSION_ID="([0-9.]*)"'
+        match = re.search(pattern, cmd_output)
+        return  match.group(1) if match else ""
+
+    __linux_version_cmds = {
+        "systemd": {"command": "cat /etc/os-release", "parser": __parse_systemd},
+        "centos": {"command": "cat /etc/centos-release", "parser": __parse_centos},
+        "amazon": {"command": "cat /etc/system-release", "parser": __parse_amazon},
+        "rhel": {"command": "cat /etc/redhat-release", "parser": __parse_rhel},
+        "oracle": {"command": "cat /etc/redhat-release", "parser": __parse_rhel},
+        "debian": {"command": "cat /etc/debian_version", "parser": lambda x: x},
+        "ubuntu": {"command": "cat /etc/lsb-release", "parser": __parse_ubuntu},
+    }
+
+    def __init__(self, inventory_path: str):
+        """LinuxAgent constructor. """
+        super().__init__(inventory_path=inventory_path)
+
+    def _get_os_version(self, platform: str) -> str:
+        cmd = LinuxAgent.__linux_version_cmds[platform]["command"]
+        parser = LinuxAgent.__linux_version_cmds[platform]["parser"]
+        cmd_output = ConnectionManager.execute_commands(self.inventory_path, cmd)
+        return parser(cmd_output.get('output')) if cmd_output.get('output', None) else ""
+
+    def get_os_version(self) -> str:
+        """Get os version from the OS platform."""
+        # detect platform
+        os_version: str = ""
+        if not (os_version := self._get_os_version('systemd')):
+            for platform in LinuxAgent.__linux_version_cmds:
+                if (os_version := self._get_os_version(platform)):
+                    logger.info(f"OS Version get using {platform} method.")
+                    break
+        else:
+            logger.info("OS Version get using systemd method.")
+        return os_version
+
+
+class WindowsAgent(WazuhAgent):
+    """Windows Agent Class."""
+
+    def get_os_version(self) -> str:
+        """Get os version from the platform."""
+
+        cmd = "[System.Environment]::OSVersion.Version.Major"
+        version_major = ConnectionManager.execute_commands(self.inventory_path, cmd)
+
+        cmd = "[System.Environment]::OSVersion.Version.Minor"
+        version_minor = ConnectionManager.execute_commands(self.inventory_path, cmd)
+
+        os_version: str = f"{version_major.get('output')}.{version_minor.get('output')}"
+
+        return os_version
+
+class MacOsAgent(WazuhAgent):
+    """Windows Agent Class."""
```

Comment:
Inconsistent, it is macOS not Windows

Reply:
Done in [4406a97](https://github.com/wazuh/wazuh-qa/pull/5407/commits/4406a973732c2ab038d91e798ad95dc9a74584fc)